src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -234,21 +245,32 @@ private String getKibanaVersion(RestRequest request) {
 	private boolean updateCache(final String user, final String token, final boolean isOperationsUser, final String kbnVersion) {
 		logger.debug("Updating the cache for user '{}'", user);
 		try{
+			// This is the key to authentication.  Before listProjectsFor, we haven't actually authenticated
+			// anything in this plugin, using the given token.  If the token is valid, we will get back a
+			// list of projects for the token.  If not, listProjectsFor will throw an exception and we will
+			// not update the cache with the user and token.  In this way we will keep bogus entries out of
+			// the cache.
 			Set<String> projects = listProjectsFor(token);
-			cache.update(user, projects, isOperationsUser);
+			cache.update(user, token, projects, isOperationsUser);
 
 			Set<String> roles = new HashSet<String>();
 			if (isOperationsUser)
 				roles.add("operations-user");
 
 			setDashboards(user, projects, roles, esClient, kibanaIndex, kbnVersion, use_cdm, cdm_project_prefix, settings);
+		} catch (KubernetesClientException if8e) {
```

Comment:
I would name this var 'e' since its valid in the scope and more along the lines of convention